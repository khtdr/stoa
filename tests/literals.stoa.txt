--- literal values ---

string
0
3.14
42
true
false
nil

--- truthiness ---
"": true
"string": true
42: true
0: true
3.14: true
0.00: true
true: true
false: false
nil: false
