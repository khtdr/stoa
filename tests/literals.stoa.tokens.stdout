PRINT[3,1]
STRING[3,7]
SEMICOLON[3,31]
PRINT[4,1]
STRING[4,7]
SEMICOLON[4,9]
PRINT[5,1]
STRING[5,7]
SEMICOLON[5,15]
PRINT[6,1]
NUMBER[6,7]
SEMICOLON[6,8]
PRINT[7,1]
NUMBER[7,7]
SEMICOLON[7,11]
PRINT[8,1]
NUMBER[8,7]
SEMICOLON[8,9]
PRINT[9,1]
TRUE[9,7]
SEMICOLON[9,11]
PRINT[10,1]
FALSE[10,7]
SEMICOLON[10,12]
PRINT[11,1]
NIL[11,7]
SEMICOLON[11,10]
PRINT[13,1]
STRING[13,7]
SEMICOLON[13,9]
PRINT[14,1]
STRING[14,7]
SEMICOLON[14,27]
PRINT[15,1]
STRING[15,7]
PLUS[15,14]
BANG[15,16]
BANG[15,17]
STRING[15,18]
SEMICOLON[15,20]
PRINT[16,1]
STRING[16,7]
PLUS[16,20]
BANG[16,22]
BANG[16,23]
STRING[16,24]
SEMICOLON[16,32]
PRINT[17,1]
STRING[17,7]
PLUS[17,14]
BANG[17,16]
BANG[17,17]
NUMBER[17,18]
SEMICOLON[17,20]
PRINT[18,1]
STRING[18,7]
PLUS[18,13]
BANG[18,15]
BANG[18,16]
NUMBER[18,17]
SEMICOLON[18,18]
PRINT[19,1]
STRING[19,7]
PLUS[19,16]
BANG[19,18]
BANG[19,19]
NUMBER[19,20]
SEMICOLON[19,24]
PRINT[20,1]
STRING[20,7]
PLUS[20,16]
BANG[20,18]
BANG[20,19]
NUMBER[20,20]
SEMICOLON[20,24]
PRINT[21,1]
STRING[21,7]
PLUS[21,16]
BANG[21,18]
BANG[21,19]
TRUE[21,20]
SEMICOLON[21,24]
PRINT[22,1]
STRING[22,7]
PLUS[22,17]
BANG[22,19]
BANG[22,20]
FALSE[22,21]
SEMICOLON[22,26]
PRINT[23,1]
STRING[23,7]
PLUS[23,15]
BANG[23,17]
BANG[23,18]
NIL[23,19]
SEMICOLON[23,22]
